lib/gmp/gmp.nit

```diff
@@ -0,0 +1,398 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Multi precision integer and rational number using gmp lib
+module gmp
+
+private import native_gmp
+
+redef class Numeric
+
+    # The BigInt equivalent of `self`
+    fun to_bi: BigInt do return self.to_i.to_bi
+
+    # The Ratio equivalent of `self`
+    fun to_r: Ratio do return self.to_f.to_r
+
+end
+
+redef class Text
+
+    # Is `self` a well-formed BigInt (i.e. parsable via `to_bi`)
+    #
+    #     assert "123".is_bi
+    #     assert "-123".is_bi
+    #     assert not "0b1011".is_bi
+    #     assert not "123u8".is_bi
+    #     assert not "Not an BigInt".is_bi
+    fun is_bi: Bool do
+        var pre = prefix("-")
+        if pre != null then
+            return pre.text_after.is_dec
+        else
+            return is_dec
+        end
+    end
+
+    # Is `self` a well-formed Ratio (i.e. parsable via `to_r`)
+    #
+    #     assert "123".is_r
+    #     assert "-123".is_r
+    #     assert "1/2".is_r
+    #     assert "-1/2".is_r
+    #     assert not "-1/-2".is_r
+    #     assert not "0b1011".is_r
+    #     assert not "123u8".is_r
+    #     assert not "Not an Ratio".is_r
+    fun is_r: Bool do
+        var frac = split_once_on('/')
+        if frac.length == 2 then
+            return frac[0].is_bi and frac[1].is_dec
+        else
+            return is_bi
+        end
+    end
+
+    # If `self` contains a BigInt, return the corresponding BigInt
+    #
+    #     assert("123".to_bi == 123.to_bi)
+    #     assert("-123".to_bi == -123.to_bi)
+    fun to_bi: BigInt do
+        assert is_bi
+        var tmp = new NativeMPZ
+        tmp.set_str(self.to_cstring, 10i32)
+        return new BigInt(tmp)
+    end
+
+    # If `self` contains a Ratio, return the corresponding Ratio
+    #
+    #     assert("123".to_r == 123.to_r)
+    #     assert("-123".to_r == -123.to_r)
+    #     assert("1/2".to_r == 0.5.to_r)
+    #     assert("-1/2".to_r == -0.5.to_r)
+    fun to_r: Ratio do
+        assert is_r
+        var tmp = new NativeMPQ
+        tmp.set_str self.to_cstring
+        return new Ratio(tmp)
+    end
+end
+
+redef class Float
+    redef fun to_bi do
+        var tmp = new NativeMPZ
+        tmp.set_d self
+        return new BigInt(tmp)
+    end
+
+    redef fun to_r do
+        var tmp = new NativeMPQ
+        tmp.set_d self
+        return new Ratio(tmp)
+    end
+end
+
+redef class Int
+    redef fun to_bi do
+        var tmp = new NativeMPZ
+        tmp.set_si self
+        return new BigInt(tmp)
+    end
+
+    redef fun to_r do
+        var tmp = new NativeMPQ
+        tmp.set_si(self, 1)
+        return new Ratio(tmp)
+    end
+end
+
+# Multi precision Integer numbers.
```

Comment:
Same comment here. Since this is an important class, it should be documented in more details.